CONTRIBUTING.md

```diff
@@ -151,30 +136,72 @@ Once you've run this, your repository should be set for most changes (i.e., you
 
 ### 4. Make the changes you've discussed
 
-Try to keep the changes focused to the issue. We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted. 
+Try to keep the changes focused to the issue.  
+We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted.
+Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub 
+workflow when making changes.
+[This guide][link_gitworkflow] provides a useful overview for this workflow.
+Before making a new branch, make sure your master is up to date.
 
-Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub workflow when making changes. [This guide][link_gitworkflow] provides a useful overview for this workflow.
+```
+git checkout master
+git fetch upstream master
+git merge upstream/master
+```
+
+Then, make your new branch.
+
+```
+git checkout -b MYBRANCH
+```
+
+As you're making changes, make sure your branch is kept up to date with 
+
+```
+git fetch upstream master
+git merge upstream/master
+```
+
+If you know what rebasing is, please only use it for changes that haven't been pushed.
+If you don't know what rebasing is, don't do it at all, as it is the easiest way to make your 
+repository a disaster zone.
+Please make sure to review the `tedana` [style conventions](#style-guide) and test your changes.
+
+If you are new to ``git``, there are several GUI git clients that you may find helpful, such as
+- [GitKraken][link_git_kraken]
+- [GitHub Desktop][link_github_desktop]
+- [SourceTree][link_source_tree]
 
-Before creating your pull request, please make sure to review the `tedana` [style conventions](#style-guide).
 
 ### 5. Test your changes
```

Comment:
One general thing -- I think the way this section is currently written risks discouraging people from opening pull requests "early and often" which is something we explicitly want.

Especially since the testing is so minimal, I'd almost rather we didn't include this section and instead wrote something about how we expect all new code to include tests, and guides for writing good tests. WDYT ?

Reply:
That sounds like a good alternative. I can see why the phrasing I had is discouraging, I'll amend it ahead of Thanksgiving.

Reply:
@emdupre do you consider this resolved now since you approved earlier?